File: Java/프로그래머스/3/49191. 순위/순위.java
class Solution {
    public int solution(int n, int[][] results) {
        int answer = 0;
        int[][] arr = new int[n + 1][n + 1];

        for (int i = 0; i < results.length; i++) {
            int win = results[i][0];
            int lose = results[i][1];
            arr[win][lose] = 1;
            arr[lose][win] = -1;
        }

        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                for (int k = 1; k <= n; k++) {
                    if (arr[i][k] == 1 && arr[k][j] == 1) {
                        arr[j][i] = -1;
                        arr[i][j] = 1;
                    }
                    if (arr[i][k] == -1 && arr[k][j] == -1) {
                        arr[j][i] = 1;
                        arr[i][j] = -1;
                    }
                }
            }
        }

        for (int[] ints : arr) {
            int count = 0;
            for (int i : ints) {
                if (i == 1 || i == -1) count++;
            }
            if (count == n - 1) answer++;
        }

        return answer;
    }
}

File: Python/프로그래머스/3/49191. 순위/순위.py
def solution(n, results):
    answer = 0
    arr = [[0] * (n + 1) for _ in range(n + 1)]
    for a, b in results:
        arr[a][b] = 1
        arr[b][a] = -1

    for i in range(1, n + 1):
        for j in range(1, n + 1):
            for k in range(1, n + 1):
                if arr[i][k] == 1 and arr[k][j] == 1:
                    arr[i][j] = 1
                    arr[j][i] = -1
                if arr[i][k] == -1 and arr[k][j] == -1:
                    arr[i][j] = -1
                    arr[j][i] = 1

    for lst in arr:
        cnt = 0
        for i in lst:
            if i == 1 or i == -1:
                cnt += 1
        if cnt == n - 1:
            answer += 1

    return answer